pymatgen/io/vasp/inputs.py

```diff
@@ -1384,18 +1384,25 @@ def from_str(cls, string: str) -> Self:
 
         # Automatic gamma and Monk KPOINTS, with optional shift
         if style in {"g", "m"}:
-            kpts = [int(i) for i in lines[3].split()]
-            kpts_shift = (0, 0, 0)
+            kpts = tuple(int(i) for i in lines[3].split())
+            assert len(kpts) == 3
+
+            kpts_shift: tuple[float, float, float] = (0, 0, 0)
             if len(lines) > 4 and coord_pattern.match(lines[4]):
                 with contextlib.suppress(ValueError):
-                    kpts_shift = [float(i) for i in lines[4].split()]
+                    _kpts_shift = tuple(float(i) for i in lines[4].split())
+                if len(_kpts_shift) == 3:
+                    kpts_shift = _kpts_shift
 
             return cls.gamma_automatic(kpts, kpts_shift) if style == "g" else cls.monkhorst_automatic(kpts, kpts_shift)
 
         # Automatic kpoints with basis
         if num_kpts <= 0:
             _style = cls.supported_modes.Cartesian if style in "ck" else cls.supported_modes.Reciprocal
-            kpts_shift = [float(i) for i in lines[6].split()]
+            _kpts_shift = tuple(float(i) for i in lines[6].split())
+            if len(_kpts_shift) == 3:
+                kpts_shift = _kpts_shift
```

Comment:
This fix looks terrible. A cleaner fix would be much appreciated.

Reply:
thanks! i'll take a look at the last error and this fix tomorrow

Reply:
That would be much appreciated. Previous error in 8583e39b10cc8aaeeba98f1ff880817962f2e14f has been fixed in 2d5e0bb889ab7ccdfd0b7d3c5c65c8333145ed07. The only error is related to this (`Kpoint`) as well, see [my comments](https://github.com/materialsproject/pymatgen/pull/3705#issuecomment-2024658023).